### PR TITLE
Accessibility improvements - form fields

### PR DESCRIPTION
**Pull request by robinmolen**
Closes open-formulieren/open-forms#4716

These are some accessibility changes regarding chapters 1.3.1 and 1.3.5 of the accessibility rapport

### CODE REVIEW

File: src/components/forms/Checkbox/Checkbox.js
Comment (sergei-maertens) [inline src/components/forms/Checkbox/Checkbox.js:47]: isn't it more foo-proof to compute the error message ID once and pass it down to the component entirely, instead of only passing the `inputId`?

```jsx
const errorMessageId = invalid ? `${id}-error-message` : undefined;

<Component
  aria-describedby={errorMessageId}
>
    {touched && <ValidationErrors error={error} id={errorMessageId} />}
</Component>
```

File: src/components/forms/EmailField/EmailField.js
Comment (sergei-maertens) [inline src/components/forms/EmailField/EmailField.js:14]:
```suggestion
  autocomplete: PropTypes.oneOf(['email']),
```
it doesn't make sense to allow other autocomplete strings, does it?

File: src/components/forms/RadioField/RadioField.js
Comment (sergei-maertens) [inline src/components/forms/RadioField/RadioField.js:38]: left over TODO

**Review by sergei-maertens — COMMENTED** [review]
more review for the formio components will follow!

File: src/components/forms/Checkbox/Checkbox.js
Comment (robinmolen) [reply src/components/forms/Checkbox/Checkbox.js:47]: Ahh.. yeah that would be better... Will fix

File: src/components/forms/EmailField/EmailField.js
Comment (robinmolen) [reply src/components/forms/EmailField/EmailField.js]: Ah check, yeah i guess so. Or at least, for now that is the correct solution. I've changed it 👍

File: src/components/forms/ValidationErrors.js
Comment (sergei-maertens) [inline src/components/forms/ValidationErrors.js:5]: I think you want to leave this `undefined` instead of defaulting to an empty string so that if no `id` prop is passed, it doesn't get set. Otherwise this could possibly result in multiple divs with `<div id="">...</div>` and then IDs are not unique in the page.

The HTML 5 spec also says:

> When specified on [HTML elements](https://html.spec.whatwg.org/multipage/infrastructure.html#html-elements), the [id](https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute) attribute value must be unique amongst all the [IDs](https://dom.spec.whatwg.org/#concept-id) in the element's [tree](https://dom.spec.whatwg.org/#concept-tree) and must contain at least one character. The value must not contain any [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace).

So unless React does something clever here, I'd stick to an explicit `id = undefined`

File: src/formio/utils.js
Comment (sergei-maertens) [inline src/formio/utils.js:20]: Instead of passing `parentElement`, I think it makes more sense to pass the `this` object (which is the component) instance and then you can use `component.refs.messageContainer` - that's how Formio's ref mechanism is supposed to be used.

File: src/formio/components/FileField.js
Comment (sergei-maertens) [inline src/formio/components/FileField.js:295]: this should also use `this.refs.fileBrowse`

File: src/formio/templates/file.ejs
Comment (sergei-maertens) [inline src/formio/templates/file.ejs:1]: This looks odd in the Chromatic visual regression tests, especially on mobile there are now table headers but the rows/files themselves don't use this same layout.

I'll also need some more explanation on why the change to table is required :thinking:

**Review by sergei-maertens — COMMENTED** [review]
I'd like to see some (Jest) tests that assert the aria-describedby behaves as expected. See all the annotations about missing test coverage. I'm also very interested in how `multiple: true` components behave!

File: src/components/forms/ValidationErrors.js
Comment (robinmolen) [reply src/components/forms/ValidationErrors.js:5]: Ah yeah fair point.. Yeah you're right, that shouldn't be an empty string 👍

File: src/formio/components/FileField.js
Comment (robinmolen) [reply src/formio/components/FileField.js:295]: ah okay, yeah if we can reach it via `this` then that would be better. I'll change it 👍

**Comment by robinmolen** [comment]
Formio doesn't validate inputs inside a component with the `multiple` property individually. Meaning: if we have a textfield component with the `multiple` property, where the first input is valid and the second invalid, we cannot determine in the code which input is the valid/invalid one. This means we cannot add things like `aria-invalid="true"` to a specific input inside the component (because, again, we don't know which input is invalid).

So to prevent some weird issues for screenreader users, we cannot make components with the `multiple` property accessible. The accessibility change to link the error message to the input (with `aria-describedby`) and to add `aria-invalid="true"` won't happen for components with the `multiple` property

File: src/formio/templates/file.ejs
Comment (robinmolen) [reply src/formio/templates/file.ejs:1]: As we discussed, the `aria-labelledby` also provides the needed information to screenreaders. I've changed the table back to a list

File: src/formio/templates/file.ejs
Comment (sergei-maertens) [inline src/formio/templates/file.ejs:9]: `strong` is typically used for its semantic meaning, and that has pretty much made `b` obsolete since that's only meant for visual presentation, which you can (and should) do with CSS.

So if the `strong` tag is an a11y problem, then we should remove it and apply the relevant/proper CSS instead of replacing it with `b` through a class name.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/formio/templates/file.ejs
Comment (robinmolen) [reply src/formio/templates/file.ejs:9]: Ah shit, i thought i removed those...

Yep, will replace them

File: src/formio/utils.js
Comment (robinmolen) [reply src/formio/utils.js]: I've changed it to:

```
setErrorAttributes(..., ... ,..., this.refs.messageContainer.id);
```

Seeing how this function only needs the id of the message container

File: src/formio/templates/file.ejs
Comment (robinmolen) [reply src/formio/templates/file.ejs:9]: Ah yeah, fair enough

File: src/formio/templates/file.ejs
Comment (robinmolen) [reply src/formio/templates/file.ejs:21]: Mhh the apple screenreader didn't have any issues with the `title`, but its indeed better to be save 👍 

And if found what you described for the translations with variables. I hadn't considered different language structures...

File: src/formio/components/Checkbox.spec.js
Comment (sergei-maertens) [inline src/formio/components/Checkbox.spec.js:62]: you can use [`toHaveClass`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohaveclass)

File: src/formio/components/jest-util.js
Comment (sergei-maertens) [inline src/formio/components/jest-util.js:1]: I think this will be mega-confusing with getting a formio component instance from the `WebForm`. Maybe rename it?

```suggestion
const getComponentNode = element => element.closest('[ref="component"]');
```

File: src/formio/components/Checkbox.spec.js
Comment (robinmolen) [reply src/formio/components/Checkbox.spec.js:65]: Ah yeah thats a lot nicer (and probably better...). I'll change it

File: src/formio/components/jest-util.js
Comment (robinmolen) [reply src/formio/components/jest-util.js]: Mhh yeay, you might be right.. getComponentNode sounds a lot better.. 👍

**Review by sergei-maertens — APPROVED** [review]
Can be merged once #722 is merged!

**Comment by sergei-maertens** [comment]
The test failures are known issues and will be fixed via #723

**Pull request merged**
